src/docfx/validation/CustomRuleProvider.cs

```diff
@@ -254,9 +256,18 @@ private static ErrorLevel ConvertSeverity(ValidationSeverity severity)
 
         private Dictionary<string, ValidationRules>? GetValidationRules(SourceInfo<string> rules)
         {
-            return !string.IsNullOrEmpty(rules.Value)
-                ? JsonUtility.DeserializeData<Dictionary<string, ValidationRules>>(_fileResolver.ReadString(rules), rules.Source?.File)
-                : null;
+            try
+            {
+                return !string.IsNullOrEmpty(rules.Value)
```

Comment:
Instead of catching in the inner method, how about catching on the outer layer instead?
This time, it fails on JSON deserialization, it could fail with other things next time.
We can think about the build pipeline, where we should not fail to let the build continue.

For instance, in the `[DocsetBuilder](https://github.com/dotnet/docfx/blob/39856aad02896115625e38763b6d36be7bb07bee/src/docfx/build/DocsetBuilder.cs#L90)`, when we can continue without these dependencies generated?